packages/sanity/src/core/studio/components/navbar/search/components/SearchPopover.tsx

```diff
@@ -71,179 +64,63 @@ export function SearchPopover({
-  }, [hasValidTerms, recentSearches.length, result.hits.length])
-
-  if (!open) {
-    return null
-  }
-
-  return (
-    <Portal>
-      <FocusLock autoFocus={false} disabled={disableFocusLock} returnFocus>
-        <Overlay style={{zIndex}} />
-
-        <CommandListProvider
-          ariaActiveDescendant={itemIndices.length > 0}
-          ariaChildrenLabel={hasValidTerms ? 'Search results' : 'Recent searches'}
-          ariaHeaderLabel="Search results"
-          autoFocus
-          data-testid="search-results-popover"
-          initialSelectedIndex={hasValidTerms ? lastActiveIndex : 0}
-          itemIndices={itemIndices}
-        >
-          <SearchPopoverContent
-            filtersVisible={filtersVisible}
-            hasValidTerms={hasValidTerms}
-            onClose={handleClose}
-            open={open}
-            position={position}
-          />
-        </CommandListProvider>
-      </FocusLock>
-    </Portal>
-  )
-}
-
-function SearchPopoverContent({
-  filtersVisible,
-  hasValidTerms,
-  onClose,
-  open,
-  position,
-}: {
-  filtersVisible: boolean
-  hasValidTerms: boolean
-  onClose: (lastActiveIndex: number) => void
-  open: boolean
-  position: PopoverPosition
-}) {
-  const [popoverElement, setPopoverElement] = useState<HTMLDivElement | null>(null)
-  const {isTopLayer, zIndex} = useLayer()
-  const {scheme} = useColorScheme()
-  const {getTopIndex} = useCommandList()
-
-  /**
-   * Store top-most search result scroll index on close
-   */
-  const handleClose = useCallback(() => {
-    onClose(getTopIndex())
-  }, [getTopIndex, onClose])
-
   /**
    * Check for top-most layer to prevent closing if a portalled element (i.e. menu button) is active
    */
   const handleClickOutside = useCallback(() => {
-    if (open && isTopLayer) {
-      handleClose()
+    if (isTopLayer && onSearchClose && open) {
+      onSearchClose()
     }
-  }, [handleClose, isTopLayer, open])
+  }, [isTopLayer, onSearchClose, open])
 
   useClickOutside(handleClickOutside, [popoverElement])
 
-  /**
-   * Bind hotkeys to close action
-   */
-  useSearchHotkeys({onClose: handleClose, open})
-
   return (
-    <SearchPopoverCard
-      $position={position}
-      overflow="hidden"
-      radius={POPOVER_RADIUS}
-      ref={setPopoverElement}
-      scheme={scheme}
-      shadow={2}
-      style={{zIndex}}
-    >
-      <SearchHeader onClose={handleClose} />
-
-      {filtersVisible && (
-        <FiltersCard borderTop>
-          <Filters />
-        </FiltersCard>
-      )}
-
-      <CommandListContainer>
-        {hasValidTerms ? (
-          <SearchResults onClose={handleClose} />
-        ) : (
-          <RecentSearches showFiltersOnClick />
-        )}
-      </CommandListContainer>
-    </SearchPopoverCard>
+    <SearchWrapper hasValidTerms={hasValidTerms} onClose={onClose} onOpen={onOpen} open={open}>
+      <Portal>
+        <FocusLock autoFocus={false} disabled={disableFocusLock} returnFocus>
```

Comment:
There is some focus issue with the search text input. When I start typing, the focus is lost when the search results popover appear and the search text input expands. Using `autoFocus` seems to solve the problem, but that might introduce other issues?

Reply:
This focus issue seems to be related to "Recent Searches". When I have recent searches, focus works as expected.

Reply:
Great spot, in global search we have one shared input used across two command lists.

The issue here was that `autoFocus` was being set on command lists – however, it's possible to have NO visible command lists, e.g. if you have no recent search results OR search query

Changes:
- In global search: `autoFocus` is now set on the search header itself and not on the individual command lists (we check for touch support so we don't auto focus on mobile to avoid the software keyboard popping up un-asked)
- Command lists will no longer check for touch support when calling `autoFocus`. They shouldn't have been opinionated in this way to begin with